### DISCUSSION TITLE
I get "--dev-url cannot be empty" on `atlas schema apply -u ... -:to schema.sql`

### DISCUSSION THREAD

**Discussion by 205g0** [ariga/atlas#1761] [Q&A] [answered]
Pretty much the title and I'd like to migrate the the db in `-u` to the state in `schema.sql`, -u is my dev database. The migration from dev to prod is not handled by Atlas.

If I add `--dev-url` the same connection string as in `-u` I get: _Error: sql/migrate: taking database snapshot: sql/migrate: connected database is not clean: found table "products" in schema "shop"_

Guess I've misunderstood something...

**Comment by a8m** [comment]
Hey @205g0,

Can you please paste here the full command you ran? `schema inspect` is used for inspection.

**Reply by 205g0** [reply]
Sure, I've meant "apply", sorry, editing the title and the full command is:

```bash
atlas schema apply --to file://schema.sql -u $MY_DEV_DB
```

My use case is adding  a column to a existing table.

**Reply by a8m** [reply, marked as answer]
Got it, check out my comment there: https://github.com/ariga/atlas/issues/1644#issuecomment-1596830101

TL;DR - The `-u` is meant to be the database you want to apply the schema on, and the `--dev-url` is a local dev-database that is used for validating, normalizing, and computing the state defined by the `schema.sql` file. You can change the command to the following:
```bash
atlas schema apply \
  --to file://schema.sql \
  -u $MY_DEV_DB \
  --dev-url "docker://mysql/8/dev"
```
Or:
```bash
atlas schema apply \
  --to file://schema.sql \
  -u $MY_DEV_DB \
  --dev-url "docker://postgres/15/dev?search_path=public"
```

**Reply by 205g0** [reply]
@a8m apologes for my replies, I've been on mobile yesterday and missed that one link. Now, everything works and I've grokked the concept!

However, why can Atlas not just use the respective docker string as default after it dtetected the db type? Moreover, the flag itself is a bit misleasing, it's not my dev db but rather a validation db/engine for Atlas (e.g. --val-url)
